### PR TITLE
Add cameras and players to map by double clicking

### PR DESCRIPTION
**Pull request by hungnguyen94**
This makes it possible to add cameras and players by double clicking on the map. It will popup a dialog, allowing you to select either a player or a camera. 
With this, we can omit the coordinates fields in the add player/camera dialog. 

I also moved the `uibModal` methods to a separate service.

### CODE REVIEW

**Comment by ChrisBerg** [comment]
Nice work on the FabricMapDialogs!

File: src/main/webapp/app/map/map-table-view-template.html [hunk 1]
```diff
@@ -40,7 +40,7 @@
         </td>
     </tr>
 
-    <tr class="btn btn-raised btn-xs" ng-click="vm.addCamera()">
+    <tr class="btn btn-raised btn-xs" ng-click="vm.addCamera(vm.project.map)">
```
Comment (ChrisBerg) [inline src/main/webapp/app/map/map-table-view-template.html]: The camera is now created in the corner where it almost falls of the map, could this be created somewhere else?

File: src/main/webapp/app/map/map-table-view-template.html [hunk 2]
```diff
@@ -83,7 +83,7 @@
         </td>
     </tr>
 
-    <tr class="btn btn-raised btn-xs" ng-click="vm.addPlayer()">
+    <tr class="btn btn-raised btn-xs" ng-click="vm.addPlayer(vm.project.map)">
```
Comment (ChrisBerg) [inline src/main/webapp/app/map/map-table-view-template.html]: Same for map

File: src/main/webapp/app/map/fabric-map.directive.js [hunk 3]
```diff
@@ -77,7 +81,7 @@
              * Initializes the map with a canvas and loads the cameras and players.
              */
             function init() {
-                var canvas = new fabric.Canvas(element[0]);
+                var canvas = new fabric.CanvasEx(element[0]);
                 canvas.selection = false;
                 scope.canvas = canvas;
                 scope.canvas.add(label);
```
Comment (ChrisBerg) [inline src/main/webapp/app/map/fabric-map.directive.js]: Could this label be removed on mouse click? It now stays at the old position when dragging a camera or a player, which seems kind of odd.

**Comment by ChrisBerg** [comment]
Works really well, nice work! One small remark: could you set the focus of the add camera dialog and add player dialog form to the "Name" field?

This could be done for example at the `CameraDialogController` by changing `angular.element('.form-group:eq(1)>input').focus();` to `angular.element('.form-group:eq(0)>input').focus();` (line 16 in /app/map/camera-dialog.controller.js)

**Comment by trombomart** [comment]
Nothing to comment besides Chris' comments. Nice job!

**Comment by hungnguyen94** [comment]
`index.html` is actually needed here, because the `fabric.canvasex.js` file is not injected by `gulp`, but has to be manually loaded. I don't commit unnecessary files :wink:

**Comment by cadw** [comment]
Looks great to me, it is mergeable in my eyes.

File: src/main/webapp/app/map/map-table-view-template.html (hunk 1 above)
Comment (hungnguyen94) [inline src/main/webapp/app/map/map-table-view-template.html]: Are you adding cameras by clicking on the `add camera` button on the table? 
You can now add cameras by double clicking. The button on the table should probably be removed.

File: src/main/webapp/app/map/fabric-map.directive.js (hunk 3 above)
Comment (hungnguyen94) [inline src/main/webapp/app/map/fabric-map.directive.js]: Ideally, the label should move with the camera or player. But for now, I'll see if I can remove it.

**Comment by hungnguyen94** [comment]
Label moves with the cameras and players now. I changed the focus of the camera dialog to the `name` field. I also removed the `Add player` and `Add camera` button from the table view.

**Comment by hungnguyen94** [comment]
image
Highlighting is made a bit more clear.

**Comment by ChrisBerg** [comment]
@hungnguyen94 nice, i am getting real thirsty right now...

**Comment by hungnguyen94** [comment]
So many merge commits :cry:

**Comment by ChrisBerg** [comment]
Nice, i think it is mergeable after all checks pass.

**Pull request merged**